[onewire] IndexOutOfBoundsException and iButton issues

I try to switch over from onewire1 to onewire2.

If I define my things with a name and a location, I get an error message:
```
	Thing basic iButton_Ulrich "Schlüssel Ulrich" @ "Flur EG"
	[
		id="01.05D98A1A0000",
		refresh=2
	]
	{
		Channels:
			Type present : Present
	}
```
```
03-Okt-2019 13:23:03.852 [DEBUG] [internal.owserver.OwserverConnection] - read pid 1667 -> connection still alive
03-Okt-2019 13:23:03.919 [DEBUG] [internal.owserver.OwserverConnection] - failed requesting messageType READ, size 46, controlFlags 0x00000104, payload '/01.05D98A1A0000/type'->return code -1, size 24, controlFlags 0x00000104, payload '' [I/O error: exception while reading packet - Read timed out]
03-Okt-2019 13:23:03.951 [DEBUG] [ternal.handler.OwserverBridgeHandler] - updating thing properties for onewire:basic:Flur:iButton_Ulrich failed: I/O error: exception while reading packet - Read timed out, adding to end of list
03-Okt-2019 13:23:04.984 [ERROR] [ternal.handler.OwserverBridgeHandler] - refresh encountered exception of class java.lang.IndexOutOfBoundsException: Index: 0, Size: 0, please report bug
```

Without name and location the error is gone:
```
Thing basic iButton_Ulrich
	[
		id="01.05D98A1A0000",
		refresh=2
	]
	{
		Channels:
			Type present : Present
	}
```
```
03-Okt-2019 13:35:08.650 [DEBUG] [internal.owserver.OwserverConnection] - read pid 1667 -> connection still alive
03-Okt-2019 13:35:08.718 [DEBUG] [internal.owserver.OwserverConnection] - failed requesting messageType READ, size 46, controlFlags 0x00000104, payload '/01.05D98A1A0000/type'->return code -1, size 24, controlFlags 0x00000104, payload '' [I/O error: exception while reading packet - Read timed out]
03-Okt-2019 13:35:08.761 [DEBUG] [ternal.handler.OwserverBridgeHandler] - updating thing properties for onewire:basic:Flur:iButton_Ulrich failed: I/O error: exception while reading packet - Read timed out, adding to end of list
```
In both cases, the iButton isn't connected.

If I disconnect the iButton I get this this message in the event log
`2019-10-03 13:43:57.102 [hingStatusInfoChangedEvent] - 'onewire:basic:Flur:iButton_Ulrich' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): slave missing
`
and when I connect it it is this message:
`2019-10-03 13:47:29.587 [hingStatusInfoChangedEvent] - 'onewire:basic:Flur:iButton_Ulrich' changed from OFFLINE (COMMUNICATION_ERROR): slave missing to ONLINE
`
I onewire1 I had a property ignoreReadErrors, which doesn't seem to be available anymore, but could solve the messages above.

Nevertheless I expected a status change at the item belonging to the thing channel. But this isn't happening.
`Switch  swi_iButton_Ulrich      "Ulrichs Schlüssel [%s]"                   <keyring>     (gLog) { channel="onewire:basic:Flur:iButton_Ulrich:Present" }
`

I'm using openhab 2.5.0 M3 on Raspbian GNU/Linux 9 (stretch).

--- a/bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwBaseThingHandler.java
+++ b/bundles/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/handler/OwBaseThingHandler.java
@@ -139,10 +139,6 @@ public abstract class OwBaseThingHandler extends BaseThingHandler {
 
         refreshInterval = configuration.refresh * 1000;
 
-        if (thing.getChannel(CHANNEL_PRESENT) != null) {
-            showPresence = true;
-        }
-
         // check if all required properties are present. update if not
         for (String property : requiredProperties) {
             if (!properties.containsKey(property)) {
@@ -187,6 +183,10 @@ public abstract class OwBaseThingHandler extends BaseThingHandler {
         } catch (OwException e) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
             return;
+        }
+
+        if (thing.getChannel(CHANNEL_PRESENT) != null) {
+            showPresence = true;
         }
 
         validConfig = true;

